scripts/download_osm_data.py

```diff
@@ -106,36 +106,36 @@ def convert_iso_to_geofk(
     store_path_data = Path.joinpath(Path().cwd(), "data", "osm")
     country_list = country_list_to_geofk(snakemake.params.countries)
 
-    eo.get_osm_data(
+    eo.save_osm_data(
         primary_name="power",
         region_list=country_list,
         feature_list=["substation", "line", "cable", "generator"],
         update=False,
         mp=True,
         data_dir=store_path_data,
+        out_dir=store_path_data,
```

Comment:
Sorry @GbotemiB 
I've just seen that out_dir and data_dir are the same.
out_dir should be the store_path_resources.
Could you revise that? by doing so the download_osm_data part is really up-to-date&state-of-the-art

Reply:
Thanks for picking that out. I made a mistake when I was committing the changes

Reply:
> Actually @davide-f, I think it is that way because of the structure from earth-osm https://github.com/pypsa-meets-earth/earth-osm/blob/main/earth_osm/eo.py#L128-L138

No, that is the default value but can be changed (I implemented that feature XD)
The idea is to disentangle where the raw files are saved (data_dir) from where the output files are saved (out_dir).
If we use out_dir=data_dir download_osm_data cannot operate in parallel for different countries, which is performance-wise terrible.
Enabling to distinguish the 2 folders can in principle allow that, as the output folder can be in the form "resources/{scenario_name}/..."

Reply:
I don't quite understand.

There may be a mix-up.  The data_dir is the location for the pbf files, while the out_dir handles the files relating to cable, generators, lines, and substations.

Reply:
> I don't quite understand.
> 
> There may be a mix-up.  The data_dir is the location for the pbf files, while the out_dir handles the files relating to cable, generators, lines, and substations.

Yeah, that's exactly the point.
They don't need to be the same folder. The pbfs can be in one folder and results in another one.

The paralleling problem I mentioned is the following:
Let's assume we use the scenario management for NG and KE and run the current download_osm_data in parallel.

Earth_osm can download the 2pbfs in parallel and store it in data_dir.
However, I'd out_dir is the same as data_dir, the 2 instances will try to write on the same temporary files in data_dir.

Instead, if we use a different out_dir different than out_dir (the res dir), this problem doesn't occur

Reply:
Oh I see. I think it is clear now.

Reply:
With the suggestion you gave, this is what the folder structure looks like

```bash
resources
└── SD_BJ
    └── osm
        └── raw
            └── out
                ├── BJ_SD_generator.csv
                ├── BJ_SD_generator.geojson
                ├── BJ_SD_line.csv
                ├── BJ_SD_line.geojson
                ├── BJ_SD_substation.csv
                └── BJ_SD_substation.geojson
```

```bash
data/osm
├── pbf
│   ├── benin-latest.osm.pbf
│   ├── benin-latest.osm.pbf.md5
│   ├── sudan-latest.osm.pbf
│   └── sudan-latest.osm.pbf.md5
└── power
    ├── BJ_power.json
    └── SD_power.json
```

Reply:
Sounds good :)
As a comment, the "SD_BJ" folder, applies only if the user specify a scenario name.

Reply:
Yeah. You are right.

But there are changes from the way it used to work. Before the `out` folder in the `resources` tree used to be under the `data/osm` tree. But now the case is different.

Reply:
The out folder may be placed into resources/osm/raw/out and then results are moved 1 folder backwards.
If that simplifies, we can keep it that way.
Alternatively, we can place it directly into resources/osm/raw (no out folder); this may be cleaner but may requires some checks with shutil.move to avoid issues if the source and destination paths match.

Reply:
I have made some adjustment to the script as you suggested. The results are now moved 1 folder backwards.

Reply:
Fantastic!